source/scheduler.cpp

```diff
@@ -15,6 +15,7 @@ bool Scheduler::tick() {
         std::move(const_cast<std::unique_ptr<Event>&>(m_events.top()));
     m_events.pop();
     event->operator()();
+    m_current_event_local_time = event->get_time();
```

Comment:
I think we should swap this two lines
```suggestion
    m_current_event_local_time = event->get_time();
    event->operator()();
```